common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.cpp

```diff
@@ -15,6 +15,110 @@ Point RNSScreenShadowNode::getContentOriginOffset(
   return {contentOffset.x, contentOffset.y};
 }
 
+    std::optional<std::reference_wrapper<const ShadowNode::Shared>>
+    findHeaderConfigChild(
+            const YogaLayoutableShadowNode &screenShadowNode)  {
+        for (const ShadowNode::Shared &child : screenShadowNode.getChildren()) {
+            if (std::strcmp(
+                    child->getComponentName(), "RNSScreenStackHeaderConfig") == 0) {
+                return {std::cref(child)};
+            }
+        }
+        return {};
+    }
+
+    static constexpr const char *kScreenDummyLayoutHelperClass =
+            "com/swmansion/rnscreens/utils/ScreenDummyLayoutHelper";
+
+#ifdef ANDROID
+        std::optional<float> findHeaderHeight(
+                const int fontSize,
+                const bool isTitleEmpty) {
+            JNIEnv *env = facebook::jni::Environment::current();
+
+            if (env == nullptr) {
+                LOG(ERROR) << "[RNScreens] Failed to retrieve env\n";
+                return {};
+            }
+
+            jclass layoutHelperClass = env->FindClass(kScreenDummyLayoutHelperClass);
+
+            if (layoutHelperClass == nullptr) {
+                LOG(ERROR) << "[RNScreens] Failed to find class with id "
+                           << kScreenDummyLayoutHelperClass;
+                return {};
+            }
+
+            jmethodID computeDummyLayoutID =
+                    env->GetMethodID(layoutHelperClass, "computeDummyLayout", "(IZ)F");
+
+            if (computeDummyLayoutID == nullptr) {
+                LOG(ERROR)
+                        << "[RNScreens] Failed to retrieve computeDummyLayout method ID";
+                return {};
+            }
+
+            jmethodID getInstanceMethodID = env->GetStaticMethodID(
+                    layoutHelperClass,
+                    "getInstance",
+                    "()Lcom/swmansion/rnscreens/utils/ScreenDummyLayoutHelper;");
+
+            if (getInstanceMethodID == nullptr) {
+                LOG(ERROR) << "[RNScreens] Failed to retrieve getInstanceMethodID";
+                return {};
+            }
+
+            jobject packageInstance =
+                    env->CallStaticObjectMethod(layoutHelperClass, getInstanceMethodID);
+
+            if (packageInstance == nullptr) {
+                LOG(ERROR)
+                        << "[RNScreens] Failed to retrieve packageInstance or the package instance was null on JVM side";
+                return {};
+            }
+
+            jfloat headerHeight = env->CallFloatMethod(
+                    packageInstance, computeDummyLayoutID, fontSize, isTitleEmpty);
+
+            return {headerHeight};
+        }
+#endif // ANDROID
```

Comment:
Let's move these methods below `appendChild`

Reply:
@tboba I'm afraid those methods have to be declared before they're used in cpp

Reply:
@alduzy But didn't we achieve different order, but in RNSScreenComponentDescriptor.h? 😄 
It's not a big deal to have them here though, so if it's really not possible, I think we can proceed with what we have right now.